storage/core/src/main/java/com/hortonworks/streamline/storage/impl/jdbc/JdbcStorageManager.java

```diff
@@ -61,14 +63,17 @@ public JdbcStorageManager(QueryExecutor queryExecutor) {
     @Override
     public void add(Storable storable) throws AlreadyExistsException {
         log.debug("Adding storable [{}]", storable);
-        final Storable existing = get(storable.getStorableKey());
-
-        if(existing == null) {
-            addOrUpdate(storable);
-        } else if (!existing.equals(storable)) {
-            throw new AlreadyExistsException("Another instance with same id = " + storable.getPrimaryKey()
-                    + " exists with different value in namespace " + storable.getNameSpace()
-                    + ". Consider using addOrUpdate method if you always want to overwrite.");
+
+        try {
+            queryExecutor.insert(storable);
+        } catch (StorageException ex) {
+            if (ex.getCause() instanceof SQLIntegrityConstraintViolationException) {
+                throw new AlreadyExistsException("Another instance with same id = " + storable.getPrimaryKey()
```

Comment:
That is another topic, there was a todo added in [StorageException](https://github.com/hortonworks/streamline/blob/master/storage/core/src/main/java/com/hortonworks/streamline/storage/exception/StorageException.java#L18). It seems it was added in very early days(mid 2015) . It seems storage and cache APIs require some cleanup.

I guess we are trying to give better error message for the client without showing JDBC provider's error message as that looks to be confusing. This is not a critical issue to be addressed but trying to give possible comprehensible error message to user.